src/main/scala/TableGroup.scala

```diff
@@ -1,6 +1,412 @@
+        )
+        csvwPropertyType match {
+          case PropertyType.Annotation => annotations += (property -> newValue)
+          case PropertyType.Common     => commonProperties += (property -> newValue)
+          case PropertyType.Inherited =>
+            inheritedProperties += (property -> newValue)
+          case _ => {
+            warnings :+= ErrorMessage(
+              "invalid_property",
+              "metadata",
+              "",
+              "",
+              property,
+              ""
+            )
+          }
+        }
+      }
+      ()
+    }
+    (annotations, commonProperties, inheritedProperties, warnings)
+  }
+
+  private def createTableObjectsAndSetWarnings(
+      tableGroupNode: ObjectNode,
+      baseUrl: String,
+      lang: String,
+      commonProperties: Map[String, JsonNode],
+      inheritedProperties: Map[String, JsonNode]
+  ): (Map[String, Table], Array[ErrorMessage]) = {
+    var warnings = Array[ErrorMessage]()
+
+    tableGroupNode.path("tables") match {
+      case t: ArrayNode if t.isEmpty() =>
+        throw new MetadataError("Empty tables property")
+      case t: ArrayNode =>
+        extractTablesFromNode(
+          t,
+          baseUrl,
+          lang,
+          commonProperties,
+          inheritedProperties
+        )
+      case n if n.isMissingNode => throw new MetadataError("No tables property")
+      case _                    => throw new MetadataError("Tables property is not an array")
+    }
+  }
+
+  def extractTablesFromNode(
+      tablesArrayNode: ArrayNode,
+      baseUrl: String,
+      lang: String,
+      commonProperties: Map[String, JsonNode],
+      inheritedProperties: Map[String, JsonNode]
+  ): (mutable.Map[String, Table], Array[ErrorMessage]) = {
+    var warnings = Array[ErrorMessage]()
+    val tables = Map[String, Table]()
+    for (tableElement <- tablesArrayNode.elements().asScalaArray) {
+      tableElement match {
+        case tableElementObject: ObjectNode => {
+          var tableUrl = tableElement.get("url")
+          if (!tableUrl.isTextual) {
+            warnings = warnings :+ ErrorMessage(
+              "invalid_url",
+              "metadata",
+              "",
+              "",
+              s"url: $tableUrl",
+              ""
+            )
+            tableUrl = new TextNode("")
+          }
+          tableUrl = new TextNode(
+            new URL(new URL(baseUrl), tableUrl.asText()).toString
+          )
+          tableElementObject.set("url", tableUrl)
+          val table = Table.fromJson(
+            tableElementObject,
+            baseUrl,
+            lang,
+            commonProperties,
+            inheritedProperties
+          )
+          tables += (tableUrl.asText -> table)
+        }
+        case _ => {
+          warnings = warnings :+ ErrorMessage(
+            "invalid_table_description",
+            "metadata",
+            "",
+            "",
+            tableElement.toPrettyString,
```

Comment:
So here in the case block `tableElement` is not an object. It is some kind of JsonNode and  `toPrettyString` is a function thats available on jsonNode.

Reply:
I was just suggesting we provide a message to the user letting them know why the warning has shown up. Presumably this will just show a warning with the value and a message saying `invalid_table_description`?

Might be nice to extend this message to have more info.